src/vs/base/common/async.ts

```diff
@@ -23,13 +23,16 @@ export function createCancelablePromise<T>(callback: (token: CancellationToken)
 
 	const thenable = callback(source.token);
 	const promise = new Promise<T>((resolve, reject) => {
-		source.token.onCancellationRequested(() => {
+		const subscription = source.token.onCancellationRequested(() => {
+			subscription.dispose();
```

Comment:
also `source.dispose()`

Reply:
But then it is exactly as with the disposable store that you were commenting about?

Reply:
sure, but without the extra store object